src/librustc/util/profiling.rs

```diff
@@ -93,16 +93,27 @@ macro_rules! define_categories {
                 $(
                     let (hits, total) = self.query_counts.$name;
 
+                    //normalize hits to 0%
+                    let hit_percent =
+                        if total > 0 {
+                            ((hits as f32) / (total as f32)) * 100.0
+                        } else {
+                            0.0
```

Comment:
You could use Laplace smoothing instead - `(1 + hits) / (2 + total)`.
It's really simple, but still useful when you want to count almost any kind of `interesting/all` fractions.

Reply:
I'm intrigued but I don't think I completely understand the idea. In the case where the category has no data `hits=0` and `total=0` so wouldn't this result in `(1 + 0) / (2 + 0) => 1/2 => 0.5` aka 50% cache hits when there were no hits or attempts at all?

Reply:
Yeah, if there's zero prior knowledge then the cache hit probability is 50% - it's either hit or not :)

If you have some better prior knowledge than that, then you can tweak the [`α` parameter](https://en.wikipedia.org/wiki/Additive_smoothing) in `(α + hits)/(2α + total)`.
Either way you won't have to divide by zero.

Reply:
It just seems like a bug to me that if there's no data, cache hit ratio would be 50%. If it's ok with you, I'd prefer to leave the code as is. :)